llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -517,6 +517,17 @@ static Decomposition decompose(Value *V,
       return Result;
     }
 
+    // shl nsw x, shift is mul nsw x, (1<<shift),
```

Comment:
nit: might be good to wrap the expressions in `()` or something else to separate them from the interleave text